### DISCUSSION TITLE
Is there a way to pass command-line parameters to helm via. flux?

### DISCUSSION THREAD

**Discussion by TonyNoble** [fluxcd/flux2#4234] [Q&A]
Case in point.  The grafana helm chart contains this snippet in the templates:
```
{{- define "grafana.ingress.apiVersion" -}}
{{- if and ($.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19-0" .Capabilities.KubeVersion.Version) }}
{{- print "networking.k8s.io/v1" }}
{{- else if $.Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
{{- print "networking.k8s.io/v1beta1" }}
{{- else }}
{{- print "extensions/v1beta1" }}
{{- end }}
{{- end }}
```

When defining a helm release, this always resolves to `extensions/v1beta1`, because the Kubernetes client available to helm (via flux) has no knowledge of what's installed in the cluster *and doesn't validate by default*.

This can be remedied (I believe) if calling helm directly, by passing the '--validate' command line parameter to helm.  But as far as my google searching has lead me, I can't see a way to do this in the HelmRelease definition.

Does anyone have any pointers for this?  Or do I need to disable ingress in the grafana chart and deploy my own custom definition alongside the HelmRelease?

**Comment by makkes** [comment]
This is the first time I'm hearing that HelmReleases don't support querying API capabilities. 🤔

**Reply by TonyNoble** [reply]
@makkes - my understanding was that helm relies on the k8s client to provide the information, which may not actually query the cluster unless specifically told to do so - https://github.com/helm/helm/issues/10760

I do appreciate I may have missed something obvious though, so if someone can create a HelmRelease of grafana (with ingress enabled) and get a different result to me, I'd be very grateful for any enlightenment :)

**Reply by makkes** [reply]
helm-controller isn't using the equivalent of `helm template`, it's instrumenting the Helm SDK to install a release so you have all the capabilities that you would have when running `helm install`, including the `Capabilities.APIVersions` [built-in object](https://helm.sh/docs/chart_template_guide/builtin_objects/).

**Reply by b4nst** [reply]
Well that's weird, we're able to reproduce the same behaviour here. We get a
```
no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1" ensure CRDs are installed first
```

for
```yaml
{{- $autoscalingv2 := .Capabilities.APIVersions.Has "autoscaling/v2" -}}
{{- if $autoscalingv2 }}
apiVersion: autoscaling/v2
{{- else }}
apiVersion: autoscaling/v2beta1
```

But 
```shell
kubectl get --raw /apis/autoscaling
```

results in
```json
{
  "kind": "APIGroup",
  "apiVersion": "v1",
  "name": "autoscaling",
  "versions": [
    {
      "groupVersion": "autoscaling/v2",
      "version": "v2"
    },
    {
      "groupVersion": "autoscaling/v1",
      "version": "v1"
    }
  ],
  "preferredVersion": {
    "groupVersion": "autoscaling/v2",
    "version": "v2"
  }
}
```

**Reply by stefanprodan** [reply]
@b4nst are you running the latest version of Flux? If so please open an issue in helm-controller about this.

**Reply by b4nst** [reply]
Tracked in https://github.com/fluxcd/helm-controller/issues/936
